process/common.go

```diff
@@ -183,3 +205,319 @@ func GetMarshalizedHeaderFromStorage(
 
 	return buffHdr, nil
 }
+
+// GetMetaHeaderWithNonce method returns a meta block header with a given nonce
```

Comment:
Please check if the duplicated code can be reduced

Reply:
Refactored